crates/sc-proof-of-time/src/state_manager.rs

```diff
@@ -547,15 +599,102 @@ impl InternalState {
 struct StateManager {
     /// The PoT state
     state: Mutex<InternalState>,
+
+    /// For AES verification
+    proof_of_time: ProofOfTime,
 }
 
 impl StateManager {
     /// Creates the state.
-    pub fn new(config: PotConfig) -> Self {
+    pub fn new(config: PotConfig, proof_of_time: ProofOfTime) -> Self {
         Self {
             state: Mutex::new(InternalState::new(config)),
+            proof_of_time,
         }
     }
+
+    /// Verify the proofs and append to the chain.
+    #[allow(clippy::too_many_arguments)]
+    fn verify_and_append(
```

Comment:
It doesn't seem appropriate for "state" to do any kind of verification. It should just consolidate pre-verified values.

Reply:
I tried other methods, but this is the cleanest approach IMO.

This wraps the whole sequence: take lock - check if it can be appended - drop lock - verify proofs outside lock - reacquire lock, append

We can add another abstraction on top that does something similar: call to check if proofs can be appended - verify proofs - call again to append. But this adds unnecessary complexity. I don't see why state manager cannot verify if needed